x-pack/plugins/aiops/public/components/log_categorization/attachments_menu.tsx

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { EuiContextMenuProps } from '@elastic/eui';
+import {
+  EuiButton,
+  EuiButtonIcon,
+  EuiContextMenu,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiPanel,
+  EuiPopover,
+  EuiSpacer,
+  EuiSwitch,
+} from '@elastic/eui';
+import type { CasesPublicStart } from '@kbn/cases-plugin/public';
+import type { Capabilities } from '@kbn/core-capabilities-common';
+import type { EmbeddableStart } from '@kbn/embeddable-plugin/public';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import type { SaveModalDashboardProps } from '@kbn/presentation-util-plugin/public';
+import {
+  LazySavedObjectSaveModalDashboard,
+  withSuspense,
+} from '@kbn/presentation-util-plugin/public';
+import React, { useCallback, useState } from 'react';
+import { useMemo } from 'react';
+import type { DataView } from '@kbn/data-views-plugin/common';
+import { EMBEDDABLE_PATTERN_ANALYSIS_TYPE } from '@kbn/aiops-log-pattern-analysis/constants';
+import { useTimeRangeUpdates } from '@kbn/ml-date-picker';
+import type { PatternAnalysisEmbeddableState } from '../../embeddables/pattern_analysis/types';
+import type { RandomSamplerOption, RandomSamplerProbability } from './sampling_menu/random_sampler';
+import { useCasesModal } from '../../hooks/use_cases_modal';
+
+const SavedObjectSaveModalDashboard = withSuspense(LazySavedObjectSaveModalDashboard);
+
+interface AttachmentsMenuProps {
+  dataView: DataView;
```

Comment:
Could this use `useAiopsAppContext` for some of the props here?

Reply:
Done in: [#c696e4e](https://github.com/elastic/kibana/pull/199478/commits/c696e4e35ab09dbcddd358f16c0da97456c74d7b)